src/OpenSage.Game/Gui/Apt/ActionScript/ObjectContext.cs

```diff
@@ -0,0 +1,145 @@
+            Item = item;
+
+            //initialize item dependent properties
+            InitializeProperties();
+        }
+
+        /// <summary>
+        /// return a variable, when not present return Undefined
+        /// </summary>
+        /// <param name="name">variable name</param>
+        /// <returns></returns>
+        public virtual Value GetMember(string name)
+        {
+            Value result;
+
+            if (!Variables.TryGetValue(name, out result))
+            {
+                Debug.WriteLine("[WARN] Undefined variable: " + name);
+                result = Value.Undefined();
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Check wether or not a string is a builtin flash function
+        /// </summary>
+        /// <param name="name">function name</param>
+        /// <returns></returns>
+        public bool IsBuiltInFunction(string name)
+        {
+            switch (name)
+            {
+                case "stop":
+                    return Item is SpriteItem;
+                case "setInterval":
+                    return Item is SpriteItem;
+                case "gotoAndPlay":
+                    return Item is SpriteItem;
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Execute a builtin function maybe move builtin functions elsewhere
+        /// </summary>
+        /// <param name="name"><function name/param>
+        public void CallBuiltInFunction(string name, List<Value> args, ActionContext context)
+        {
+            switch (Item)
+            {
+                case SpriteItem si:
+                    switch (name)
+                    {
+                        case "stop":
+                            si.Stop();
+                            break;
+                        case "setInterval":
+                            break;
+                        case "gotoAndPlay":
+                            si.Goto(args[0].ToString());
+                            si.Play();
+                            break;
+                    }
+                    break;
+                case RenderItem ri:
+                    break;
+            }
+        }
+
+        private void InitializeProperties()
+        {
+            //TODO: avoid new fancy switch
+            switch (Item.Character)
+            {
+                case Text t:
+                    Variables["textColor"] = Value.FromString(t.Color.ToHex());
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// used by text
+        /// </summary>
+        /// <param name="value">value name</param>
+        /// <returns></returns>
+        public Value ResolveValue(string value)
+        {
+            var path = value.Split('.');
+            var obj = this;
+
+            if(path.Length>1)
+            {
+                if (path[0] == "_parent")
```

Comment:
How does this logic relate to the check for "_parent" in `ActionContext.GetValue`?

Reply:
In ObjectContext it is asked for a specific variable relative to the current object, e.g. "_parent.buttonColor" or just a variable inside the current object "variable" etc. In ActionContext.GetValue it is checked for builtin variables in general. It's not quite nice yet, but i'll hopefully be able to clean the system up